### DISCUSSION TITLE
Question: How does `spacemacs|spacebind` modify behaviour of `use-package`

### DISCUSSION THREAD

**Discussion by lebensterben** [syl20bnr/spacemacs#14519] [Q&A] [answered]
It says that
https://github.com/syl20bnr/spacemacs/blob/b71e5fdb179d7fdba92f80ce07aedbfb70339411/core/core-spacebind.el#L417

And AFAIK the only way to 'hijack' an external function's behaviour is through `advice`, but `use-package` is not advised.

I'm curious on how `:spacebind` keyword is added to it.

**Comment by JAremko** [comment, marked as answer]
@lebensterben It can be used inside `use-package` form.  Tests have a case for it: https://github.com/syl20bnr/spacemacs/blob/7ef15a561a1a1f00574d30f46dd564c6d648ebfd/tests/layers/%2Bdistribution/spacemacs-base/spacebind-utest.el#L59-L69
